core/src/main/scala/org/apache/spark/HttpFileServer.scala

```diff
@@ -42,10 +42,11 @@ private[spark] class HttpFileServer(
     fileDir.mkdir()
     jarDir.mkdir()
     logInfo("HTTP File server directory is " + baseDir)
-    httpServer = new HttpServer(conf, baseDir, securityManager, requestedPort, "HTTP file server")
+    httpServer = new HttpServer(conf, baseDir, securityManager, requestedPort, "HTTP file server", Some("spark.fileserver.advertisedPort"))
```

Comment:
@tnachen before I start modifying the pr, I read that the AkkaRpcEnv is planned to be removed in favour of NettyRpcEnv in 1.6.0. Is this still planned?
If so, it might be worth for me to look at that at thesame time.